server/src/main/java/org/apache/iotdb/db/service/thrift/impl/DataNodeTSIServiceImpl.java

```diff
@@ -1196,28 +1222,55 @@ private QueryId genQueryId(long id) {
   }
 
   // check whether measurement is legal according to syntax convention
-  protected void isLegalMeasurementLists(List<List<String>> measurementLists) throws TException {
+  protected void isLegalMeasurementLists(List<List<String>> measurementLists)
+      throws MetadataException {
     if (measurementLists == null) {
       return;
     }
+    StringBuilder path = new StringBuilder(IoTDBConstant.PATH_ROOT);
     for (List<String> measurementList : measurementLists) {
-      isLegalMeasurements(measurementList);
+      for (String measurement : measurementList) {
```

Comment:
why not just call isLegalMeasurement?

Reply:
because call isLegalMeasurement needs to do PathUtil.isLegalPath() size() times. By concating string, we only need to do that once.

Reply:
the path is concat all measurement of the  two level list?

Reply:
It's only for check. Measurement is supposed to be a single node name(exclude root). For example, measurementLists is {{"a","b"},{"c","d"}} and we need to check each node name. Check each node individually is the same as check "a.b.c.d". Hope  it is clearer now.

Reply:
ok, got it.